python/cudf/cudf/core/column/column.py

```diff
@@ -1832,9 +1856,17 @@ def as_column(
                 dtype=arbitrary.dtype,
             )
         elif arb_dtype.kind in ("O", "U"):
```

Comment:
If we're handling object arrays of Intervals, we should split these two cases as we don't need to check in the array for `U`

Reply:
Sounds good, I'll change this so it only checks for objects in this case :D